p2p/conn/connection_test.go

```diff
@@ -769,3 +770,273 @@ func stopAll(t *testing.T, stoppers ...stopper) func() {
+	// utilized.
+
+	// Testcases 5-7 assess if surpassing available bandwidth maintains
+	// consistent transmission delay without congestion or performance
+	// degradation. A uniform delay across these testcases is expected.
+
+	tests := []struct {
+		name              string
+		msgSize           int           // size of each message in bytes
+		totalMsg          int           // total number of messages to be sent
+		messagingRate     time.Duration // rate at which messages are sent
+		totalDuration     time.Duration // total duration for which messages are sent
+		sendQueueCapacity int           // send queue capacity i.e., the number of messages that can be buffered
+		sendRate          int64         // send rate in bytes per second
+		recRate           int64         // receive rate in bytes per second
+	}{
+		{
+			// testcase 1
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// Ideal transmission time for 50 messages is ~1 second.
+			// SendQueueCapacity should not affect transmission delay.
+			name: "queue capacity = 1, " +
+				"total load = 50 KB, " +
+				"msg rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          1 * 50,
+			sendQueueCapacity: 1,
+			messagingRate:     20 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 2
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// Ideal transmission time for 50 messages is ~1 second.
+			// Increasing SendQueueCapacity should not affect transmission
+			// delay.
+			name: "queue capacity = 50, " +
+				"total load = 50 KB, " +
+				"traffic rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          1 * 50,
+			sendQueueCapacity: 50,
+			messagingRate:     20 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 3
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// Ideal transmission time for 50 messages is ~1 second.
+			// Increasing SendQueueCapacity should not affect transmission
+			// delay.
+			name: "queue capacity = 100, " +
+				"total load = 50 KB, " +
+				"traffic rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          1 * 50,
+			sendQueueCapacity: 100,
+			messagingRate:     20 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 4
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// The test runs for 100 messages, expecting a total transmission time of ~2 seconds.
+			name: "queue capacity = 100, " +
+				"total load = 2 * 50 KB, " +
+				"traffic rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          2 * 50,
+			sendQueueCapacity: 10,
```

Comment:
[question] the queue capacity in the test name doesn't match the `sendQueueCapacity`. Should one of them be updated?

```suggestion
			name: "queue capacity = 10, " +
				"total load = 2 * 50 KB, " +
				"traffic rate = send rate",
			msgSize:           1 * 1024,
			totalMsg:          2 * 50,
			sendQueueCapacity: 10,
```

Reply:
Fixed https://github.com/celestiaorg/celestia-core/commit/cfd525ac9c274c9368b58c293d3d32290d782d53